### PR TITLE
Add patch version upgrade testing

### PR DESCRIPTION
**Pull request by sjmiller609**
## Description

> Describe the purpose of this pull request.

To make sure "helm upgrade" works between patch versions

## 🎟 Issue(s)

Closes https://github.com/astronomer/issues/issues/1622

## 🧪  Testing

This change is to add tests

## 📸 Screenshots

> Add screenshots to illustrate the changes, if applicable.

## 📋 Checklist

- [x] The PR title is informative to the user experience

### CODE REVIEW

File: bin/functional-tests/test_versioning.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    command = \
+        "helm3 upgrade --reset-values " + \
+        f"-f {config_path} " + \
+        f"-n {namespace} " + \
+        f"--version={major}.{minor}.{patch} " + \
+        f"{release_name} " + \
+        helm_chart_path
+    print(command)
+    print(check_output(command, shell=True))
+    print("The upgrade worked!")
```
Comment (sjmiller609) [inline bin/functional-tests/test_versioning.py:103]: here is the upgrade test. it tests both downgrade and upgrade

File: bin/functional-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+This function name starts with "test_", a pytest convention to indicate this is a pytest function. Pytest function parameters are test fixtures. Since this test has a parameter 'prometheus', pytest will match this name to a pytest fixture of the same name, and provide it as an argument in this function. In this case, it will consume the following test fixture:
+
+```
+@pytest.fixture(scope='session')
+def prometheus(request):
+    yield testinfra.get_host(f'kubectl://astronomer-prometheus-0?container=prometheus&namespace=astronomer')
+```
+
+This test fixture is making use of both 'pytest' and 'testinfra'. The pytest.fixture decorator indicate to pytest that this is a fixture. testinfra is used to simplify the connection logic. We can see that the test function above makes use of the testinfra host object (the result of the get_host call that is provided by this fixture) with the 'check_output' function:
```
Comment (danielhoherd) [inline bin/functional-tests/README.md:43]:
```suggestion
This test fixture is making use of both 'pytest' and 'testinfra'. The `@pytest.fixture` decorator indicate to pytest that this is a fixture. testinfra is used to simplify the connection logic. We can see that the test function above makes use of the testinfra host object (the result of the get_host call that is provided by this fixture) with the `check_output` function:
```

File: bin/functional-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+This line:
+```
```
Comment (danielhoherd) [inline bin/functional-tests/README.md:46]: Switching the leading fence to \`\`\`python would give syntax highlighting

File: bin/functional-tests/test_versioning.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
```
Comment (danielhoherd) [inline bin/functional-tests/test_versioning.py:1]: IMHO this should be run through `black` to clean up the formatting. We could add black to pre-commit, which would make it easy for us all to forget about how to format our code in favor of having it automatically formatted.

```yaml
  - repo: https://github.com/psf/black
    rev: 20.8b1
    hooks:
      - id: black
```

**Review by danielhoherd — APPROVED** [review]
Looks great! I left a few small improvement suggestions in case you want to do them.

File: bin/functional-tests/test_versioning.py (hunk 4 above)
Comment (sjmiller609) [reply bin/functional-tests/test_versioning.py:1]: Some pathing situation I think I need to get black configured correctly to find at the path. I'm going to skip for now

**Pull request merged**